lib/galaxy/jobs/runners/kubernetes.py

```diff
@@ -107,15 +111,25 @@ def queue_job(self, job_wrapper):
             "spec": self.__get_k8s_job_spec(job_wrapper)
         }
 
-        # Checks if job exists
+        # Checks if job exists and is trusted, or if it needs re-creation.
         job = Job(self._pykube_api, k8s_job_obj)
-        if job.exists():
+        if job.exists() and not self._galaxy_instance_id:
+            # if galaxy instance id is not set, then we don't trust matching jobs and we simply delete and
+            # re-create the job
+            log.debug("Matching job exists, but Job is not trusted, so it will be deleted and a new one created.")
             job.delete()
-        # Creates the Kubernetes Job
-        # TODO if a job with that ID exists, what should we do?
-        # TODO do we trust that this is the same job and use that?
-        # TODO or create a new job as we cannot make sure
-        Job(self._pykube_api, k8s_job_obj).create()
+            while job.exists():
+                sleep(3)
```

Comment:
Probably better to have some sort of maximum timeout here right? Sleep for half a second 30 times and then fail

Reply:
Agreed, that is a good idea. What would you suggest should be the behaviour when that happens? to proceed with creation of the new job at the risk of the older error, possibly with some catching, or set the job to a failure state because it couldn't be restarted?

Reply:
@ilveroluca any comments on what should be the behaviour after the timeout?

Reply:
I think it would be reasonable to set a failure state.  We can try to pick up old jobs on a best effort basis.

Regarding the timeout, I'd be a little more generous than 15 seconds (say 30s or 60s).  After all, if we just restarted the cluster things might be slow because of all the concurrent activity.